iaas-common/src/main/java/io/github/cloudiator/persistance/OperatingSystemModelRepositoryJpa.java

```diff
@@ -32,4 +39,35 @@ protected OperatingSystemModelRepositoryJpa(
       TypeLiteral<OperatingSystemModel> type) {
     super(entityManager, type);
   }
+
+  @Nullable
+  @Override
+  public OperatingSystemModel findByArchitectureFamilyVersion(OperatingSystemArchitecture architecture, OperatingSystemFamily family, OperatingSystemVersion version) {
+    String queryStringWithVersion = String
```

Comment:
Creating queries: queryStringWithVersion and queryStringWithoutVersion could be moved to constructor - in that case they will be calculated only once.

Reply:
done